### ISSUE TITLE
Marksman Errors When Opening a File in Nvim: `System.IndexOutOfRangeException: Index was outside the bounds of the array.`

### ISSUE THREAD

**Issue by thw26** [artempyanykh/marksman#235]
I have marksman installed via mason in [LazyVim](https://github.com/LazyVim/LazyVim/). I opened a [discussion](https://github.com/LazyVim/LazyVim/discussions/1064) on their side to make sure I wasn't missing something with them, but when I open a markdown file I get the following error from marksman, reported through the `:LspInfo` command in Nvim:

```
  [ERROR][2023-07-05 15:54:36] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "[15:54:35 INF] <LSP Entry> Starting Marksman LSP server: {}
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "---------------------------------------------------------------------------"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "Marksman encountered a fatal error
Please, report the error at https://
  github.com/artempyanykh/marksman/issues
---------------------------------------------------------------------------
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "Marksman version: "
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "1.0.0.0
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "OS: "
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "Linux 6.1.31-2-MANJARO #1 SMP PREEMPT_DYNAMIC Sun Jun  4 12:31:46 UTC
  2023
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "Arch: "
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"
  "X64
---------------------------------------------------------------------------
"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "System.IndexOutOfRangeException: Index was outside the bounds of the array. 
  
   at Markdig.Helpers.StringLineGroup.Iterator.NextCharNewLine()
   at Markdig.Helpers.LinkHelper.TryParseLabel[T](T& lines, Boolean allowEmpty, String& label, SourceSpan& labelSpan)
   at Markdig.Helpers.
  LinkHelper.TryParseLabel[T](T& lines, String& label, SourceSpan& labelSpan)
   at Markdig.Helpers.LinkHelper.TryParseLinkReferenceDefinition[T](T& text, String& label, String& url, String& title, SourceSpan& 
  labelSpan, SourceSpan& urlSpan, SourceSpan& titleSpan)
   at Markdig.Syntax.LinkReferenceDefinition.TryParse[T](T& text, LinkReferenceDefinition& block)
   at Markdig.Parsers.ParagraphBlockParser.
  TryMatchLinkReferenceDefinition(StringLineGroup& lines, BlockProcessor state)
   at Markdig.Parsers.ParagraphBlockParser.Close(BlockProcessor processor, Block block)
   at Markdig.Parsers.BlockProcessor.
  Close(Int32 index)
   at Markdig.Parsers.BlockProcessor.CloseAll(Boolean force)
   at Markdig.Parsers.BlockProcessor.ProcessLine(StringSlice newLine)
   at Markdig.Parsers.MarkdownParser.
  ProcessBlocks(BlockProcessor blockProcessor, LineReader lineReader)
   at Markdig.Parsers.MarkdownParser.Parse(String text, MarkdownPipeline pipeline, MarkdownParserContext context)
   at Markdig.Markdown.
  Parse(String markdown, MarkdownPipeline pipeline, MarkdownParserContext context)
   at Marksman.Parser.Markdown.scrapeText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 212
   at 
  Marksman.Parser.parseText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 483
   at Marksman.Workspace.DocModule.mk(UriWith`1 id, FSharpOption`1 version, Text text) in /home/runner/ 
  work/marksman/marksman/Marksman/Workspace.fs:line 58
   at Marksman.Workspace.DocModule.tryLoad(UriWith`1 folderId, LocalPath path) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 116

  at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 370
   at Microsoft.FSharp.Core.CompilerServices.
  GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.
  MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/
  Marksman/Workspace.fs:line 383
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.
  Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Marksman.Workspace.FolderModule.collect@363- 
  4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 383
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:
  \\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:
  \\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Microso"
  [ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734    "rpc"   "/home/thw26/.local/share/nvim/mason/bin/marksman"   "stderr"    "ft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in D: 
  \\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 77
   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System.Collections.IEnumerator.MoveNext() in D:
  \\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 64
   at Microsoft.FSharp.Collections.MapTreeModule.mkFromEnumerator[a,b](IComparer`1 comparer, MapTree`2 acc, IEnumerator`1 e) in D:
  \\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 443
   at Microsoft.FSharp.Collections.MapTreeModule.ofSeq[TKey,T](IComparer`1 comparer, IEnumerable`1 c) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.
  Core\\map.fs:line 455
   at Microsoft.FSharp.Collections.FSharpMap`2.Create(IEnumerable`1 ie) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 590
   at Microsoft.FSharp.Collections.MapModule.
  OfSeq[TKey,T](IEnumerable`1 elements) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 981
   at Marksman.Workspace.FolderModule.multiFile(String name, UriWith`1 root, IEnumerable`1 docs,
  FSharpOption`1 config) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 258
   at Marksman.Workspace.FolderModule.tryLoad(FSharpOption`1 userConfig, String name, UriWith`1 folderId) in /home/
  runner/work/marksman/marksman/Marksman/Workspace.fs:line 459
   at Marksman.Server.ServerUtil.readWorkspace(FSharpOption`1 userConfig, FSharpMap`2 roots) in /home/runner/work/marksman/marksman/Marksman/
  Server.fs:line 85
   at Marksman.Server.MarksmanServer.Initialize(InitializeParams par) in /home/runner/work/marksman/marksman/Marksman/Server.fs:line 528
"
[ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734	"rpc"	"/home/thw26/.local/share/nvim/mason/bin/marksman"	"stderr"	"System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Markdig.Helpers.StringLineGroup.Iterator.NextCharNewLine()
   at Markdig.Helpers.LinkHelper.TryParseLabel[T](T& lines, Boolean allowEmpty, String& label, SourceSpan& labelSpan)
   at Markdig.Helpers.LinkHelper.TryParseLabel[T](T& lines, String& label, SourceSpan& labelSpan)
   at Markdig.Helpers.LinkHelper.TryParseLinkReferenceDefinition[T](T& text, String& label, String& url, String& title, SourceSpan& labelSpan, SourceSpan& urlSpan, SourceSpan& titleSpan)
   at Markdig.Syntax.LinkReferenceDefinition.TryParse[T](T& text, LinkReferenceDefinition& block)
   at Markdig.Parsers.ParagraphBlockParser.TryMatchLinkReferenceDefinition(StringLineGroup& lines, BlockProcessor state)
   at Markdig.Parsers.ParagraphBlockParser.Close(BlockProcessor processor, Block block)
   at Markdig.Parsers.BlockProcessor.Close(Int32 index)
   at Markdig.Parsers.BlockProcessor.CloseAll(Boolean force)
   at Markdig.Parsers.BlockProcessor.ProcessLine(StringSlice newLine)
   at Markdig.Parsers.MarkdownParser.ProcessBlocks(BlockProcessor blockProcessor, LineReader lineReader)
   at Markdig.Parsers.MarkdownParser.Parse(String text, MarkdownPipeline pipeline, MarkdownParserContext context)
   at Markdig.Markdown.Parse(String markdown, MarkdownPipeline pipeline, MarkdownParserContext context)
   at Marksman.Parser.Markdown.scrapeText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 212
   at Marksman.Parser.parseText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 483
   at Marksman.Workspace.DocModule.mk(UriWith`1 id, FSharpOption`1 version, Text text) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 58
   at Marksman.Workspace.DocModule.tryLoad(UriWith`1 folderId, LocalPath path) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 116
   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 370
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 383
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 383
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441
   at Microso"
[ERROR][2023-07-05 15:54:37] .../vim/lsp/rpc.lua:734	"rpc"	"/home/thw26/.local/share/nvim/mason/bin/marksman"	"stderr"	"ft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 77
   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 64
   at Microsoft.FSharp.Collections.MapTreeModule.mkFromEnumerator[a,b](IComparer`1 comparer, MapTree`2 acc, IEnumerator`1 e) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 443
   at Microsoft.FSharp.Collections.MapTreeModule.ofSeq[TKey,T](IComparer`1 comparer, IEnumerable`1 c) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 455
   at Microsoft.FSharp.Collections.FSharpMap`2.Create(IEnumerable`1 ie) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 590
   at Microsoft.FSharp.Collections.MapModule.OfSeq[TKey,T](IEnumerable`1 elements) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 981
   at Marksman.Workspace.FolderModule.multiFile(String name, UriWith`1 root, IEnumerable`1 docs, FSharpOption`1 config) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 258
   at Marksman.Workspace.FolderModule.tryLoad(FSharpOption`1 userConfig, String name, UriWith`1 folderId) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 459
   at Marksman.Server.ServerUtil.readWorkspace(FSharpOption`1 userConfig, FSharpMap`2 roots) in /home/runner/work/marksman/marksman/Marksman/Server.fs:line 85
   at Marksman.Server.MarksmanServer.Initialize(InitializeParams par) in /home/runner/work/marksman/marksman/Marksman/Server.fs:line 528
"
```

Is there more information I can supply?

**Comment by artempyanykh** [comment]
@thw26 thanks for reporting this! Could you share the text of the document (or better a relevant part of) where you see this issue? I'm assuming this is happening on some files but not all of them, is it?

**Comment by lhoangan** [comment]
Hi, I'm encountering a similar error. I'm running it on neovim (v0.9.4) + mason

I've installed it by `:MasonInstall marksman` and setup with

```lua
local lspconfig = require('lspconfig')
lspconfig.marksman.setup {}
```

I get `Client 1 quit with exit code 1 and signal 0` for any `*.md` file, empty or already having text.

The error log is

```sh
[START][2023-10-24 13:15:22] LSP logging initiated
[ERROR][2023-10-24 13:15:22] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"[13:15:22 INF] <LSP Entry> Starting Marksman LSP server: {}\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"---------------------------------------------------------------------------"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"Marksman encountered a fatal error\nPlease, report the error at https://github.com/artempyanykh/marksman/issues\n---------------------------------------------------------------------------\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"Marksman version: "
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"1.0.0.0\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"OS: "
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"Linux 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"Arch: "
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"X64\n---------------------------------------------------------------------------\n"
[ERROR][2023-10-24 13:15:23] .../vim/lsp/rpc.lua:734	"rpc"	"/home/hale/.local/share/nvim/mason/bin/marksman"	"stderr"	"System.IndexOutOfRangeException: Index was outside the bounds of the array.\n   at Markdig.Helpers.StringLineGroup.Iterator.NextCharNewLine()\n   at Markdig.Helpers.LinkHelper.TryParseLabel[T](T& lines, Boolean allowEmpty, String& label, SourceSpan& labelSpan)\n   at Markdig.Helpers.LinkHelper.TryParseLabel[T](T& lines, String& label, SourceSpan& labelSpan)\n   at Markdig.Helpers.LinkHelper.TryParseLinkReferenceDefinition[T](T& text, String& label, String& url, String& title, SourceSpan& labelSpan, SourceSpan& urlSpan, SourceSpan& titleSpan)\n   at Markdig.Syntax.LinkReferenceDefinition.TryParse[T](T& text, LinkReferenceDefinition& block)\n   at Markdig.Parsers.ParagraphBlockParser.TryMatchLinkReferenceDefinition(StringLineGroup& lines, BlockProcessor state)\n   at Markdig.Parsers.ParagraphBlockParser.Close(BlockProcessor processor, Block block)\n   at Markdig.Parsers.BlockProcessor.Close(Int32 index)\n   at Markdig.Parsers.BlockProcessor.CloseAll(Boolean force)\n   at Markdig.Parsers.BlockProcessor.ProcessLine(StringSlice newLine)\n   at Markdig.Parsers.MarkdownParser.ProcessBlocks(BlockProcessor blockProcessor, LineReader lineReader)\n   at Markdig.Parsers.MarkdownParser.Parse(String text, MarkdownPipeline pipeline, MarkdownParserContext context)\n   at Markdig.Markdown.Parse(String markdown, MarkdownPipeline pipeline, MarkdownParserContext context)\n   at Marksman.Parser.Markdown.scrapeText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 212\n   at Marksman.Parser.parseText(Text text) in /home/runner/work/marksman/marksman/Marksman/Parser.fs:line 483\n   at Marksman.Workspace.DocModule.mk(UriWith`1 id, FSharpOption`1 version, Text text) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 58\n   at Marksman.Workspace.DocModule.tryLoad(UriWith`1 folderId, LocalPath path) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 116\n   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 370\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441\n   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 383\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441\n   at Marksman.Workspace.FolderModule.collect@363-4.GenerateNext(IEnumerable`1& next) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 383\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 405\n   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seqcore.fs:line 441\n   at Microsoft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 77\n   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System.Collections.IEnumerator.MoveNext() in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\seq.fs:line 64\n   at Microsoft.FSharp.Collections.MapTreeModule.mkFromEnumerator[a,b](IComparer`1 comparer, MapTree`2 acc, IEnumerator`1 e) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 443\n   at Microsoft.FSharp.Collections.MapTreeModule.ofSeq[TKey,T](IComparer`1 comparer, IEnumerable`1 c) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 455\n   at Microsoft.FSharp.Collections.FSharpMap`2.Create(IEnumerable`1 ie) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 590\n   at Microsoft.FSharp.Collections.MapModule.OfSeq[TKey,T](IEnumerable`1 elements) in D:\\a\\_work\\1\\s\\src\\fsharp\\FSharp.Core\\map.fs:line 981\n   at Marksman.Workspace.FolderModule.multiFile(String name, UriWith`1 root, IEnumerable`1 docs, FSharpOption`1 config) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 258\n   at Marksman.Workspace.FolderModule.tryLoad(FSharpOption`1 userConfig, String name, UriWith`1 folderId) in /home/runner/work/marksman/marksman/Marksman/Workspace.fs:line 459\n   at Marksman.Server.ServerUtil.readWorkspace(FSharpOption`1 userConfig, FSharpMap`2 roots) in /home/runner/work/marksman/marksman/Marksman/Server.fs:line 85\n   at Marksman.Server.MarksmanServer.Initialize(InitializeParams par) in /home/runner/work/marksman/marksman/Marksman/Server.fs:line 528\n"
```

Thank you

**Comment by artempyanykh** [comment]
@lhoangan thanks for reporting this! Could you please do 2 things:
1. Could you create an empty folder, then create a markdown file inside, and open it using `nvim` with Marksman? Does it still crash?
```
# Hello
## World
```
2. If p.1 works fine, then the problem is with the contents of a particular file. Do you store your markdown files under version control (e.g. git)? If so, the easiest way to determine which document causes the crash is to copy all `.md` files into an empty dir and open it in nvim one by one; when the server crashes, you've found the culprit. 

Basically, what I need is a snippet of text that causes the crash. My hunch says it's something to do with different languages/encodings.

**Comment by lhoangan** [comment]
Hi @artempyanykh, thank you for your response. I don't know if it's something you're expecting, but *within an empty folder*, `nvim` with marksman works fine with (1) your hello, world example, and (2) a file that crashed before (now copied over to the new empty folder beforehand). That crashing file was in a git-control folder (let's call it A) and just simply run `nvim` with marksman from within A crashes it. If I delete the `.git` folder from A and it works fine. Finally, it works fine on other git-control projects (B, C, D, ...). 

So what gives? What do you suggest me to try next?

**Comment by artempyanykh** [comment]
@lhoangan thank you! I could reproduce the problem and the fix is ready.
